migrate/snapshot.go

```diff
@@ -157,10 +180,11 @@ func mangleRoot(n *node) *node {
 	}
 	newRoot.Children["1"] = n
 	etcd := n.Children["_etcd"]
-	delete(n.Children, "_etcd")
+	newEtcd := deepCopyNode(etcd, newRoot)
```

Comment:
that would normally work, but it'd also involve changing fixEtcd (which could be done) to manually create the new node.

Probably should be done that way too. I just wanted to get the PR out there first.